services/tns_submission_queue/tns_submission_queue.py

```diff
@@ -52,6 +52,618 @@ class TNSReportError(Exception):
     pass
 
 
+def find_accessible_tnsrobot_groups(tnsrobot, user, submission_request, session):
+    """Verify that the user is allowed to submit to TNS with this robot, and return the TNSRobotGroup to use for submission if so.
```

Comment:
Is this a permission check on the robot, on TNS, or on both?

Reply:
on the TNS robot groups of the TNS robot. THe docstring is misleading, I will update it